### ISSUE TITLE
IPv6 and Xbox

### ISSUE THREAD

**Issue by theianspence** [Pugmatt/BedrockConnect#25]
I have a hunch as to why this may not be working for many people. It seems if a user’s Xbox is able to use IPv6, it ignores the manual IPv4 settings and continues with the default IPv6 DNS servers. This is obviously highly problematic, as this prevents many players from joining. However I’m not sure if there’s a way to fix this. Microsoft doesn’t allow users to tweak the IPv6 settings in the Xbox for some reason, nor disable it.

I was able to test this theory. My ISP forces IPv6 down my throat, so I turned on my Xbox, went to the settings, and connected to OpenDNS’ servers, as they have a handy URL that shows if you’re connected. It definitely didn’t work. At all. Even after restarting my Xbox. It’s definitely a shame, your software and general idea is awesome! If there’s a way around this issue, let me know. It just seems as if it’s literally impossible for some people to change their DNS servers without tweaking with their router and disabling IPv6 all together.

**Comment by Waliers** [comment]
Yeah same. And since more and more ISP's are moving to IPv6 I hope some solution comes along :(

**Comment by Pugmatt** [comment]
You are definitly correctly, this is something I've suspected for a while. I don't think there's anyway around it unfortunately, unless Xbox adds support for IPv6 DNS. Thanks for testing the theory out, I haven't been able to test it myself since I don't have  a ISP or Xbox that causes this. I'll look into this more and see if I can find anything. 
My DNS server has supported IPv6 connections for a while, but Xbox doesn't seem to allow connections to it, but maybe there's a way to fix this I haven't found yet.

**Comment by theianspence** [comment]
Of course. It’s just odd. It’s like they half implemented IPv6 support. It works, but you literally can’t change any of its settings nor disable it.
From what I understand, the Nintendo Switch doesn’t have IPv6 at all, so this shouldn’t be a problem.

**Comment by Dbloecher** [comment]
Can't you just change the IPv6 DNS server on your router instead of on your Xbox as a temporary solution until Xbox adds support for custom IPv6 DNS servers? I know it would make that DNS server apply to all devices on the network if they haven't had a custom DNS server set for them, but wouldn't the rest of the devices just resort to using the secondary DNS server? I'm not sure if it works that way or not.

**Comment by Dbloecher** [comment]
This might work for others, but I can't do this because AT&T forces you to use their DNS server on their modem/router. You could plug in another router into the AT&T router and change the DNS on the second router, but that's just creates a Double NAT and an inconvenience when they could just add custom DNS support on their router. I think AT&T just want to track me with their DNS. Microsoft and AT&T are good examples of tech giants making what should be simple tasks difficult just to show their power.

**Comment by Pugmatt** [comment]
> What would be the BedrockConnect IPv6 Adress? ;)

Take a look at the last bullet point in this section in the troubleshooting: https://github.com/Pugmatt/BedrockConnect/wiki/Troubleshooting#joining-the-featured-servers-still-directs-to-the-actual-servers-and-not-the-serverlist

Had forgotten to post it here in this issue, but setting that IPv6 address in your router is the current solution around the issue.

**Comment by uploadersoasis** [comment]
> And since more and more ISP's are moving to IPv6 I hope some solution comes along :(

The solution is to support IPv6 natively.  There is no reason in 2023 to not do so especially when XBOX Live is native IPv6 and only supports IPv4 using teredo.

So, I have not tried Bedrock Connect yet, but does its DNS support IPv6, i.e. AAAA records?  What about the master server and its list?  Is it reachable via IPv6, and will it store game servers with IPv6 addresses?  If the answer to any of these questions is "no," what is the timetable for support of IPv6?
